lib/embedded_ansible.rb

```diff
@@ -185,6 +187,67 @@ def self.configure_secret_key
   end
   private_class_method :configure_secret_key
 
+  def self.load_exist_settings
+    if File.exist?(SETTINGS_FILE)
+      File.readlines(SETTINGS_FILE)
+    else
+      _log.error("Ansible Tower settings file #{SETTINGS_FILE} doesn't exist.")
+      nil
+    end
+  end
+  private_class_method :load_exist_settings
+
+  def self.update_proxy_settings
+    exist_settings = load_exist_settings
+    return nil unless exist_settings
+
+    proxy = VMDB::Util.http_proxy_uri(:embedded_ansible) || VMDB::Util.http_proxy_uri
+    proxy_settings_start = exist_settings.index("# PROXY SETTINGS\n")
+    proxy_settings_start -= 1 if proxy_settings_start
+    proxy_settings_end = exist_settings.index { |s| s =~ /NO_PROXY/ }
+    proxy_settings_end += 1 if proxy_settings_end
+    new_settings = nil
+    if proxy
+      new_proxy = proxy_setting_lines(proxy)
+      new_settings = if proxy_settings_start
+                       [*exist_settings[0..(proxy_settings_start - 1)],
+                        *new_proxy,
+                        *exist_settings[proxy_settings_end..-1]]
+                     else
+                       [*exist_settings, *new_proxy]
+                     end
+    else
+      new_settings = if proxy_settings_start
+                       [*exist_settings[0..(proxy_settings_start - 1)],
+                        *exist_settings[proxy_settings_end..-1]]
+                     else
+                       exist_settings
+                     end
+    end
```

Comment:
I feel like this is all a bit over-engineered.

I think it would be easier to read if we just deleted the lines with these keys from the file, then appended the new proxy lines to the end if the proxy is set in our settings.

I don't think this process of finding the start of the previous proxy settings is really necessary.

Pseudo code should reduce to:

```ruby
current_contents = File.read(SETTINGS_FILE)
new_contents = current.gsub(/^.*AWX_TASK_ENV\['(HTTPS?_PROXY|NO_PROXY)'\].*$/, "")
if proxy_is_set
  new_contents << proxy_info
end
File.write(SETTINGS_FILE, new_contents)
```

Reply:
Yeah, this looks better